### PR TITLE
Include articles from TRS

### PR DESCRIPTION
**Pull request by jtrussell**
I'm not signifying an intent to merge... just wafting this out there so you folks can react to it and tell me which ideas are most horrible. FWIW I imagine the first incarnation of articles to be temporary no matter what.

### CODE REVIEW

**Comment by jtrussell** [comment]
Screen Shot 2019-09-05 at 10 37 59 PM

And here's an example of whit it might look like.

**Comment by jtrussell** [comment]
Screen Shot 2019-09-05 at 11 14 06 PM

And here's an example of what it might look like.

File: next/components/article-list.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+// Kinda janky, but this is all just kinda janky
+const CATEGORY_MYTHGARD = 'Mythgard';
+
+export default function ArticleList() {
+  const [list, setList] = useState([]);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
```
Comment (alexmiddeleer) [inline next/components/article-list.js]: Not sure what useEffect affords us here. A fancy way to run an async function inside a render function?

File: next/components/article.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    timeMsg = 'Over a year ago';
+  } else if (monthsDelta > 1) {
+    timeMsg = `${monthsDelta} months ago`;
+  } else if (daysDelta > 1) {
+    timeMsg = `${daysDelta} days ago`;
+  } else if (hrsDelta > 1) {
+    timeMsg = `${hrsDelta} hours ago`;
+  } else {
+    timeMsg = 'Just now';
+  }
```
Comment (alexmiddeleer) [inline next/components/article.js]: You were very charitable not to just plop in the publish date here as a string :P

File: next/components/article.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        .article-link {
+          color: ${theme.font};
+        }
+
+        .article-date {
+          font-style: italic;
+        }
+
+        .article-divider {
```
Comment (alexmiddeleer) [inline next/components/article.js]: looks fancy. is this gradient something we can add to the theme or is it pretty article-specific?

File: next/components/article.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        }
+      `}</style>
+      <div className="article">
+        <a className="article-link" href={url}>
+          {title}
+        </a>
+        <div className="article-date">{timeMsg}</div>
+        <hr className="article-divider" />
+        <div className="article-desc">
+          <div dangerouslySetInnerHTML={{ __html: description }} />
```
Comment (alexmiddeleer) [inline next/components/article.js]: hmm... what could go wrong 😄

File: next/components/article-list.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  if (error) return <ErrorMessage message="Unable to load articles." />;
+  if (isLoading) return <div>Loading</div>;
+  if (!list.length) return <div>No recent articles.</div>;
+
+  return (
+    <div>
+      {list.map((feedItem, ix) => (
+        <Article
+          key={ix}
+          title={feedItem.title}
```
Comment (alexmiddeleer) [inline next/components/article-list.js]: all of these props should probably have name dangerousTitle, dangerousUrl, etc.?

File: next/server-routes/articles.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const request = require('request');
+const router = require('express').Router();
+
+router.get('/rss', async (req, res) => {
+  request('https://teamrankstar.com/feed').pipe(res);
```
Comment (alexmiddeleer) [inline next/server-routes/articles.js]: Do we really need this or can we force it to be client side?

File: next/components/article-list.js (hunk 1 above)
Comment (jtrussell) [reply next/components/article-list.js]: It simulates a "component did mount" lifecycle method. It'll run the first time the component is mounted to the page, not before, and only once.

File: next/components/article.js (hunk 3 above)
Comment (jtrussell) [reply next/components/article.js]: iirc the was one other spot we had a similar gradient but it's not universal.

File: next/server-routes/articles.js (hunk 6 above)
Comment (jtrussell) [reply next/server-routes/articles.js]: CORS

Comment (jtrussell) [reply next/server-routes/articles.js]: The TRS folks would have to allow it.

**Comment by jtrussell** [comment]
Agree with you on the XSS front, not super worried about it for this poc/v1 implementation though since "we" effectively own the content getting pulled in.  ¯\_(ツ)_/¯

**Comment by jtrussell** [comment]
Still a moot point until/unless the TRS feed gets cleaned up. My preference remains for static, curated content and figuring out how we do more later.

**Comment by alexmiddeleer** [comment]
👍

**Comment by jtrussell** [comment]
Shutting this down.

**Pull request closed without merging**